### PR TITLE
[TIG-587] ESLint auto-fix and top-level overrides removed

### PR DESCRIPTION
**Pull request by stephenbelyea**
## Ticket

* [TIG-587](https://app.clickup.com/t/9009201449/TIG-587)

## Description

This PR removes the error-to-warning overrides at the root `.eslintrc.json` config, updates the simpler (and automatic via `--fix`) issues, and implements the overrides only in the `pages` and `components` sub-directories - where the remaining issues occur. 

This also introduces a `prop-types/component.js` file with basic extendable `BaseComponent` and `LinkComponent` prop type objects to make mocking out each component's props simpler. 

## Testing

* Pull down and build the branch locally
* Run `npm run lint` 
* Confirm no errors show up in the issues summary
* Confirm all warnings are from the `pages` and `components` folders
* Run `npm start`
* Wait until app builds and serves at `http://localhost:3000/`
* Login and navigate around, keeping an eye on the dev console
* Confirm no issues occur or existing functionality has changed

## Screenshots

**Linter output down to 418 warnings (mostly errors) localized under pages and components:**

image

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,94 +1,92 @@
...
 
-function App (props) {
+function App(props) {
   const { loginState, updateLoginState } = props;
 
   const [currentLoginState, updateCurrentLoginState] = useState(loginState);
   const { setContentTranslations } = useContentTranslationsContext();
 
   useEffect(() => {
+    async function setAuthListener() {
```
Comment (stephenbelyea) [inline src/App.js:51]: This fixes missing dependency issue with `useEffect` deps array. Same as the `downloadTranslations` patch below.

File: src/App.js [hunk 2]
```diff
@@ -1,94 +1,92 @@
...
-    Hub.listen('auth', (data) => {
-      switch (data.payload.event) {
-        case 'signOut':
-          updateLoginState('signIn');
+  useEffect(() => {
+    const downloadTranslations = async (langCode) => {
+      let translations;
+      switch (langCode) {
+        case 'fr':
+          translations = await getAllTranslations();
```
Comment (stephenbelyea) [inline src/App.js:75]: Declaring a net-new variable within a `switch` case is _no bueno_!

File: src/views/pageContainer/PageContainer.js [hunk 3]
```diff
@@ -76,16 +78,19 @@ function PageContainer(props) {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
-  const handleMenuNavItem = (toPath = null) => {
-    if (mobileView) updateMenuState(!menuEnabled);
-    if (toPath) navigate(toPath);
-  };
+  const handleMenuNavItem = useCallback(
```
Comment (stephenbelyea) [inline src/views/pageContainer/PageContainer.js:81]: The `useCallback` hook wrapper lets us pass a function as a dependency to a `useEffect` array without risking calling the function repeatedly on every render and data change.

**Review by klancashire — COMMENTED** [review]
Everything worked as expected and no major red flags. I have a couple of errors in my console but I believe these were present already but just in case I've attached a screenshot in case they are relevant. Pls scan but otherwise approved! 

Screenshot 2024-02-06 at 10 25 04 AM

**Comment by stephenbelyea** [comment]
> Everything worked as expected and no major red flags. I have a couple of errors in my console but I believe these were present already but just in case I've attached a screenshot in case they are relevant. Pls scan but otherwise approved!
> Screenshot 2024-02-06 at 10 25 04 AM

@klancashire can you confirm which page this was and what action(s) were happening when it appeared in the console?

**Comment by stephenbelyea** [comment]
Looks like the issue @klancashire found was due to [the `useCallback` being implemented on the handler for menu open/closed state based on screen size](https://github.com/TakingITGlobal/commit2act/pull/97/files#r1474884628). Not entirely sure what the behaviour is meant to be here and how it should interact with or override the user menu toggle action, so I've reverted my change and just eslint-ignored the specific issue for now.

**Pull request merged**